typescript/packages/jumble/src/views/AuthenticationView.tsx

```diff
@@ -1,62 +1,365 @@
+import ShapeLogo from "@/assets/ShapeLogo.svg";
 import { useAuthentication } from "@/contexts/AuthenticationContext";
-import { useCallback, useRef, useState } from "react";
+import { useCallback, useEffect, useState } from "react";
+import {
+  LuArrowLeft,
+  LuKey,
+  LuKeyRound,
+  LuCirclePlus,
+  LuLock,
+  LuTextCursorInput,
+  LuCopy,
+  LuTrash2,
+  LuCheck,
+} from "react-icons/lu";
+import {
+  type StoredCredential,
+  getStoredCredential,
+  saveCredential,
+  clearStoredCredential,
+  createPasskeyCredential,
+  createPassphraseCredential,
+  getPublicKeyCredentialDescriptor,
+} from "@/utils/credentials";
 
-const BTN_STYLE=`bg-gray-50 border-2 p-2 w-full flex-1 cursor-pointer`;
-const PW_STYLE=`bg-gray-50 border-2 p-2 w-full flex-1`;
+const BTN_PRIMARY = `w-full px-4 py-2 bg-black text-white hover:bg-gray-800 disabled:opacity-50 flex items-center justify-center gap-2`;
+const LIST_ITEM = `w-full p-2 text-left text-sm border-2 border-black hover:-translate-y-[2px] hover:shadow-[4px_4px_0px_0px_rgba(0,0,0,0.5)] shadow-[1px_1px_0px_0px_rgba(0,0,0,0.3)] transition-all duration-100 ease-in-out cursor-pointer flex items-center gap-2`;
+
+type AuthMethod = "passkey" | "passphrase";
+type AuthFlow = "register" | "login";
+
+interface ErrorCalloutProps {
+  error: string;
+  onDismiss: () => void;
+}
+
+interface SuccessRegistrationProps {
+  mnemonic?: string;
+  onLogin: () => void;
+  method: AuthMethod;
+  credentialId?: string;
+}
+
+function ErrorCallout({ error, onDismiss }: ErrorCalloutProps) {
+  return (
+    <div className="bg-red-50 border-l-4 border-red-400 p-4 mb-4">
+      <div className="flex">
+        <div className="flex-1">{error}</div>
+        <button onClick={onDismiss}>×</button>
+      </div>
+    </div>
+  );
+}
+
+function SuccessRegistration({
+  mnemonic,
+  onLogin,
+  method,
+  credentialId,
+}: SuccessRegistrationProps) {
+  const [copied, setCopied] = useState(false);
+
+  const copyToClipboard = () => {
+    if (mnemonic) {
+      navigator.clipboard.writeText(mnemonic);
+      setCopied(true);
+      setTimeout(() => setCopied(false), 750);
+    }
+  };
+
+  return (
+    <div className="text-center">
+      {method === "passkey" ? (
+        <div className="mb-4">
+          <p className="mb-2">Passkey successfully registered!</p>
+          {credentialId && (
+            <p className="text-sm text-gray-500 mt-2">Key ID: ...{credentialId.slice(-4)}</p>
```

Comment:
Using the `rawId` instead of `id` wouldn't make the credential ID renderable, but I'm not sure how meaningful this ID would be anyway. I think if the passkey was parameterized (name/displayName) that could be used meaningfully ("Ben Work" | "Ben Home"), but in lieu of that, not sure if the ID needs rendered

Reply:
Also if we used name/displayName, we'd have to track that ourselves as that's not returned from any of the web authn methods.

For future work, maybe we could have `common-identity` store the PassKey info via a KeyStore? TBD